components/rendering/src/markdown/fence.rs

```diff
@@ -0,0 +1,89 @@
+use super::codeblock::Range;
+
+impl Range {
```

Comment:
Can we have only one impl block? I would probably move all of it to `codeblock.rs` at the root of the folder rather than in a sub markdown folder.

Reply:
I would move the struct here and it should be good enough